src/analysis/raster/qgsrastercalculator.cpp

```diff
@@ -176,7 +176,7 @@ QgsRasterCalculator::Result QgsRasterCalculator::processCalculation( QgsFeedback
         if ( uniqueRasterEntries[layerRef.first].raster->crs() != mOutputCrs )
         {
           QgsRasterProjector proj;
-          proj.setCrs( ref.raster->crs(), mOutputCrs );
+          proj.setCrs( ref.raster->crs(), mOutputCrs, ref.raster->dataProvider()->transformContext() );
```

Comment:
Instead of taking this from the dataProvider(), I think it'd be cleaner to add it in the QgsRasterCalculator constructor (and *sigh* deprecate the other existing ones)